flexmeasures_entsoe/tests/test_utils.py

```diff
@@ -0,0 +1,79 @@
+import pytz
+import pytest
+import pandas as pd
+import click
+from datetime import datetime, timedelta
+
+from flexmeasures_entsoe.utils import (
+    abort_if_data_incomplete,
+    parse_from_and_to_dates,
+)
+
+
+def test_abort_if_data_incomplete():
+    """
+    Tests that the function raises click.Abort if data is incomplete.
+    1. Data is complete: No exception raised.
+    2. Data is incomplete: click.Abort is raised.
+    """
+    start = pd.Timestamp("2025-01-01 00:00")
+    end = pd.Timestamp("2025-01-02 00:00")
+    resolution = pd.Timedelta(hours=1)
+
+    # Case 1: Data is complete (24 items for 24 hours)
+    complete_data = pd.DataFrame({"val": range(24)})
+    try:
+        abort_if_data_incomplete(complete_data, start, end, resolution)
+    except click.Abort:
+        pytest.fail("Function raised Abort unexpectedly on complete data")
+
+    # Case 2: Data is incomplete (20 items for 24 hours)
+    incomplete_data = pd.DataFrame({"val": range(20)})
+    with pytest.raises(click.Abort):
+        abort_if_data_incomplete(incomplete_data, start, end, resolution)
+
+
+def test_parse_from_and_to_dates():
+    """
+    Tests CLI date parsing logic:
+    1. Explicit dates are timezone-localized correctly.
+    2. 'None' defaults to tomorrow (start of day) -> day after tomorrow.
+    """
+    tz_str = "UTC"
+    tz = pytz.timezone(tz_str)
+    one_day = timedelta(days=1)
+    now = datetime.now(tz)
+
+    # Case 1: Explicit inputs
+    input_start = datetime(2025, 5, 1)
+    input_end = datetime(2025, 5, 2)
+
+    s, e = parse_from_and_to_dates(
+        from_date=input_start, until_date=input_end, country_timezone=tz_str
+    )
+
+    # I am deducting one day in all the below asserts, because of the end_date being exclusive
```

Comment:
I think it is clearer to say that we expect `assert s - e == timedelta(days=2)` because we asked for 2 days (1st and 2nd of May).

Same with the other tests - this way it is hard to read.

Reply:
Ah okay, then `e -s`, makes more sense indeed